### ISSUE TITLE
numbers are as string transfered

### ISSUE THREAD

**Issue by Ilovegym66** [ioBroker/ioBroker.hass#125]
hass Adapter Version 1.40, iobroker js-controllerl 6.0.11, Admin 7.2.26, buanet Docker V10, Node V20.
All Sensors objects with a number are coming as a string.

**Comment by Apollon77** [comment]
Add a Debug log please from start and tell some examples pleade. thank you

**Comment by Ilovegym66** [comment]
Problems are the LexPlus10 from HASS mqtt - there are all strings, no numbers (or is this an issue with the HA plugin and mosqitto?) ..:
[hass-debug.txt](https://github.com/user-attachments/files/17373842/hass-debug.txt)

**Comment by Garfonso** [comment]
The problem here is that in HASS states *always* are strings. The frontend does some magic to convert them back into numbers (and they added some backend service to cirvumvent issues from that). But the entity.state is always contains a string. It even is in their Typescript Types. I did not dig very deep into the adapter, TBH. But I had a lot of issues with that in ioBroker.lovelace, too. 

But from what I saw, this https://github.com/ioBroker/ioBroker.hass/blob/master/main.js#L270 will always be `string`. If we want to change that on the ioBroker side, the adapter would need to assume some stuff about the type of the state (a rule of thumb in Lovelace was that if it has a unit, it probably is a number).

Currently the best workaround on the iobroker side is to create an alias and convert the state from string to number, IMHO.
